backend/src/main/java/com/aadeshandreas/ailearning/ai_learning_companion/service/TextUploadRequest.java

```diff
@@ -0,0 +1,16 @@
+package com.aadeshandreas.ailearning.ai_learning_companion.service;
+/**
+ * DTO for accepting raw text input from the frontend.
+ * Used when users paste notes directly instead of uploading a PDF.
+ */
+
+public record TextUploadRequest(String text){
+    /**
+     * Validates that the text is not null or empty.
+     */
+    public TextUploadRequest {
+        if (text == null || text.trim().isEmpty()) {
+            throw new IllegalArgumentException("Text cannot be empty");
+        }
+    }
+}
```

Comment:
```suggestion
public record DocumentText(@NotBlank(message = "Text cannot be empty") String text){}
```

We can add @NotBlank to the text field, that way Java will handle the error internally and throw an error.


I think we should call it something like DocumentText to keep with the noun consistency of class names. 
It should also be moved to the model folder instead of being located inside the service folder